hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/IncrSourceHelper.java

```diff
@@ -130,7 +130,7 @@ public static QueryInfo generateQueryInfo(JavaSparkContext jssc, String srcBaseP
       }
     });
 
-    String previousInstantTime = beginInstantTime;
+    String previousInstantTime = DEFAULT_BEGIN_TIMESTAMP;
```

Comment:
Should this be the commit time before the last commit time referenced in the checkpoint (`beginInstantTime`) if I understand your statement correctly?  Using `DEFAULT_BEGIN_TIMESTAMP`/`000` here fetches all commits before the last commit time.

Reply:
yes previousIntantTime is suppose to point to a commit before beginInstantTime. The code following this line will set the correct previous value. This default is to handle the case when beginInstantTime == 000

Reply:
OK.  Could you add some docs here on different scenarios, how `previous`, `start`, and `end` instants for `QueryInfo` are decided?

Reply:
Basically, the only change here is when `beginInstantTime === <first commit>`, the `previousInstantTime` is `DEFAULT_BEGIN_TIMESTAMP` now whereas `<first commit>` before this PR.

Reply:
Have you also gone through the logic for `MissingCheckpointStrategy.READ_LATEST` and see if there's any gap?

Reply:
I think there's still a hole here.  If `beginInstantTime` is the first commit in the active timeline and there are archived commits, `previousInstantTime` should not be set to `DEFAULT_BEGIN_TIMESTAMP`?  Could you check this case?

Reply:
Spoke to @nsivabalan about archival commits and we do not need to read archival is what I learnt.